mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/model/spi/MemberPropertyHandle.java

```diff
@@ -23,15 +20,7 @@
 	private final Member member;
 	private final MethodHandle getter;
 
-	public MemberPropertyHandle(String name, Field field) throws IllegalAccessException {
-		this( name, field, MethodHandles.lookup().unreflectGetter( field ) );
-	}
-
-	public MemberPropertyHandle(String name, Method method) throws IllegalAccessException {
-		this( name, method, MethodHandles.lookup().unreflect( method ) );
-	}
-
-	private MemberPropertyHandle(String name, Member member, MethodHandle getter) {
+	public MemberPropertyHandle(String name, Member member, MethodHandle getter) {
```

Comment:
The `MethodHandle` doesn't have a proper `toString()`, `equals()` and `hashCode()`?

If we could avoid storing the `Method`, it would be nice. From what I did on HV, it consumes a significant amount of memory.

Obviously, it only works if the `MethodHandle` provides what we are looking for for these 3 methods.

Reply:
OK, too bad...